src/components/dashboard/Dashboard.tsx

```diff
@@ -15,11 +15,26 @@
 // along with Superblocks Lab.  If not, see <http://www.gnu.org/licenses/>.
 
 import React, { Component } from 'react';
-import ProjectList from './projectList';
-import Topbar from './topbar';
 import { IProject } from '../../models';
 import style from './style.less';
-import { LoginModal } from '../modals';
+import {userService} from '../../services/user.service';
+import { Loading } from '../common';
+import Loadable from 'react-loadable';
+import Topbar from './topbar';
+
+const ProjectList = Loadable({
+    loader: () => import(/* webpackChunkName: "ProjectList" */'./projectList'),
+    loading: Loading,
+});
+
+const LoginModal = Loadable({
+    loader: () => import(/* webpackChunkName: "LoginModal" */'../modals'),
```

Comment:
Why wrapping modals in loaders?

Reply:
I can answer my own question here. This is to simply delay the modal loading until we actually do need it (so we can reduce the bundle).

Just read this from their own documentation :)

"Modals, tabs, and many more UI components hide content until the user has done something to reveal it."